### ISSUE TITLE
internal::WireFormat::WireTypeForField fails to default proto3 repeated fields to packed

### ISSUE THREAD

**Issue by jskeet** [google/protobuf#493]
WireFormat::MakeTag is implemented as:

```
return WireFormatLite::MakeTag(field->number(), WireTypeForField(field));
```

`WireTypeForField` in turn uses this check for "packed-ness":

```
if (field->options().packed())
```

That does *not* consider a repeated packable field in a proto3 file to be packed by default, whereas

```
if (field->is_packed())
```

does. I've made that simple change to `WireTypeForField` in my local clone, and can submit a PR if requested - but I don't know whether there's some *intended* reason for this behaviour.

**Comment by haberman** [comment]
It's probably just an oversight. It was only recently that we changed the proto3 default to packed. I think the PR would be welcome!

**Comment by xfxyjwf** [comment]
Adding @TeBoring as he implemented the new packed behavior.

**Comment by jskeet** [comment]
Okay. Will get to that on Monday then - will need to psych myself up for implementing my first unit test in C++ ;)

**Comment by jskeet** [comment]
Okay, I've got a commit here:
https://github.com/jskeet/protobuf/commit/2b593171e12394a65f9897c86dd7553f4a46d166

It's not a pull request, however, as it breaks other tests. It's not clear to me whether that just
shows that those tests are broken, or whether there's something more subtle going on.
I'd suggest that someone more familiar with C++ than myself should take it from here :)

I can work around this in my C# codegen for the moment.

**Comment by TeBoring** [comment]
The logic for paring proto3 packed repeated primitive fields is not right currently. Packed/Unpacked cannot be compatibly parsed. I will send a PR to fix it.

**Comment by nataliencl** [comment]
what does this mean. My phone has been hacked into and I cant get any help
from Sprint or Google

**Comment by TeBoring** [comment]
Fixed in #500. @haberman There is a jruby test failed. Could you take a look?
